### DISCUSSION TITLE
NNUE <> ONNX conversions

### DISCUSSION THREAD

**Discussion by JSeam2** [official-stockfish/Stockfish#4983] [Ideas]
NNUE seems to be a unique standard for chess engines, it might make sense to have a way of porting ONNX models over, vice versa.

This way it’s much easier to incorporate advances in deep learning that are usually done via tensorflow or PyTorch over, via ONNX which is widely adopted by most deep learning frameworks.

It’s a nice to have as it could potentially speed up the search for better deep learning models for the engine.

**Comment by winapiadmin** [comment]
this isn't LeelaChessZero 

bonus: NNUE is fast enough and ONNX models are not very beneficial, and it will clutter release binaries
